src/Store/Reducers/SystemCvesStore.js

```diff
@@ -56,33 +53,31 @@ function fulfilledVulnerabilities(state, action) {
                 payload,
                 isLoading: false
             },
-            prevLoadedRows: unionBy(action.payload.data, state.prevLoadedRows, 'id'),
             ...state.isAllExpanded && { expandedRows: action.payload.data.map(({ id }) => ({ id, isOpen: true })) }
         };
     }
 
     return state;
 }
 
-// #TODO refactor to use selectRows helper
 function selectEntity(state, action) {
     let newState = state;
     let selectedCves = newState.selectedCves.slice();
-    let prevLoadedRows = [].concat(state.prevLoadedRows);
 
     if (Array.isArray(action.payload)) {
         selectedCves = action.payload;
     } else {
-        (selectedCves.includes(action.payload) && selectedCves.splice(selectedCves.indexOf(action.payload), 1)) ||
+        const theSelectedCve = selectedCves.find(selectedCve => selectedCve.id === action.payload.id);
```

Comment:
I think this is more readable
`const isCveSelected = selectedCves.some(selectedCve => selectedCve.id === action.payload.id);`

`.some` returns a boolean so the intent is clearer

Reply:
I would do that, but I actually need that element to remove it from the `selectedCves` and I don't want to go through through that array again. I should probably use `findIndex` already here. So I can reuse it two rows below.